docs/superpowers/plans/2026-04-20-fake-cloud-shell.md

```
@@ -0,0 +1,507 @@
+## Local debug: viewing cloud-hosting-only UI in self-hosting
+
+Run the frontend dev server with `VITE_MOCK_CLOUD=true` to activate the fake cloud shell — no file edits, no revert checklist.
+
+```bash
+VITE_MOCK_CLOUD=true npm run dev --workspace=frontend
+```
+
+The app renders with a fake cloud navbar on top (with a "MOCK" badge), a hardcoded fake project, all `isCloud` gates true, and cloud-only data pages (Deployments / Backup / Instance) in empty state. Stop and restart the dev server without the flag to return to normal self-hosting.
+
+See `docs/superpowers/specs/2026-04-20-fake-cloud-shell-design.md` for known gaps (notably: code paths that depend on postMessage from a real cloud parent do not execute in mock mode).
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)